evennia/contrib/mapbuilder.py

```diff
@@ -0,0 +1,256 @@
+                exit = create_object(exits.Exit, key="north", 
+                                aliases=["n"], location=location[0], 
+                                destination=destination[0])
+                                
+            #east
+            if destination[1] == x+1 and destination[2] == y:
+                exit = create_object(exits.Exit, key="east", 
+                                aliases=["e"], location=location[0], 
+                                destination=destination[0])                 
+            #south
+            if destination[1] == x and destination[2] == y+1:
+                exit = create_object(exits.Exit, key="south", 
+                                aliases=["s"], location=location[0], 
+                                destination=destination[0])
+                                
+            #west
+            if destination[1] == x-1 and destination[2] == y:
+                exit = create_object(exits.Exit, key="west", 
+                                aliases=["w"], location=location[0], 
+                                destination=destination[0])   
+
+
+from django.conf import settings
+from evennia.utils import utils
+import imp
+
+COMMAND_DEFAULT_CLASS = utils.class_from_module(settings.COMMAND_DEFAULT_CLASS)
+                                
+class CmdMapBuilder(COMMAND_DEFAULT_CLASS):
+    """
+    Build a map from a 2D ASCII map.
+
+    Usage:
+        @mapbuilder <path.to.module.MAPNAME>
+
+    Example:
+        @mapbuilder evennia.contrib.mapbuilder.EXAMPLE_MAP
+        @mapbuilder world.gamemap.MAP
+
+    This is a simple way of building a map of placeholder or otherwise
+    bare rooms from a 2D ASCII map. The command merely imports the map
+    and runs the build_map function in evennia.contrib.mapbuilder.
+    This function should be altered with keys and instructions that
+    suit your individual needs.
+    """
+    key = "@mapbuilder"
+    aliases = ["@buildmap"]
+    locks = "cmd:superuser()"
+    help_category = "Building"
+
+    def func(self):
+        "Starts the processor."
+
+        caller = self.caller
+        args = self.args
+        map = None
+        
+        #Check if arguments passed.
+        if not args:
+            caller.msg("Usage: @mapbuilder <path.to.module.VARNAME>")
+            return
+        
+        #Breaks down path into PATH, VARIABLE
+        args = args.rsplit('.', 1)
+        
+        try:
+            #Retrieves map variable from module.
+            map = utils.variable_from_module(args[0],args[1])
+            
+        except Exception as exc:
+            #Or relays error message if fails.
+            caller.msg(exc)
+        
+        #Display map retrieved.
+        caller.msg("Creating Map...") 
+        caller.msg(map)
+        
+        #Pass map to the bulid function.
+        build_map(caller, map)
+        caller.msg("Map Created.")                              
+
+#Helper function for readability.
+def prepare_map(map):
```

Comment:
Maybe call this `map_to_list` to make it clearer what it does? Also, as a helper function used in `build_map` and `CmdMapBuilder`, it should be moved up to be just above `build_map` in the source file. If it's not to be used outside this module but is only meant as a helper, it's also praxis to start its name with an underscore.